Services/CO.CDP.Forms.WebApi/UseCase/UpdateFormSectionAnswersUseCase.cs

```diff
@@ -23,8 +23,52 @@ public async Task<bool> Execute((Guid formId, Guid sectionId, Guid answerSetId,
         var section = await formRepository.GetSectionAsync(formId, sectionId)
             ?? throw new UnknownSectionException($"Unknown section {sectionId} in form {formId}.");
 
-        var sharedConsent = await formRepository.GetSharedConsentDraftWithAnswersAsync(formId, organisationId)
-                            ?? CreateSharedConsent(organisation, section.Form);
+        var sharedConsent = await formRepository.GetUntrackedSharedConsent(formId, organisationId);
+        if (sharedConsent == null)
+        {
+            sharedConsent = CreateSharedConsent(organisation, section.Form);
+
+            await UpdateOrAddStandardAnswers(formRepository, answerSetId, answers, section, sharedConsent);
+            await formRepository.SaveSharedConsentAsync(sharedConsent);
+
+            return true;
+        }
+
+        if (sharedConsent.SubmissionState == Persistence.SubmissionState.Submitted)
+        {
+            CleanSharedConsent(sharedConsent);
+
+            await UpdateOrAddStandardAnswers(formRepository, answerSetId, answers, section, sharedConsent);
+            await formRepository.SaveSharedConsentAsync(sharedConsent);
+        }
+        else
+        {
+            sharedConsent = await formRepository.GetSharedConsentWithAnswersAsync(formId, organisationId);
```

Comment:
This should query for a draft shared consent.

Reply:
The reason for this is we only ever want to update draft shared consents. There's absolutely no reason NOT to include this condition here. There's a risk of mistakenly updating non-draft shared consent if the condition isn't here.

Reply:
yep, make sense with refactored logic.